daemon/src/noise.rs

```diff
@@ -6,15 +6,19 @@ use tokio::net::TcpStream;
 
 pub static NOISE_MAX_MSG_LEN: u32 = 65535;
 pub static NOISE_TAG_LEN: u32 = 16;
-// TODO: investigate what these params do and whether we can use get authentication for free using
-// them
-static NOISE_PARAMS: &str = "Noise_XX_25519_ChaChaPoly_BLAKE2s";
+static NOISE_PARAMS: &str = "Noise_IK_25519_ChaChaPoly_BLAKE2s";
 
-pub async fn initiator_handshake(connection: &mut TcpStream) -> Result<TransportState> {
+pub async fn initiator_handshake(
+    connection: &mut TcpStream,
+    local_priv_key: x25519_dalek::StaticSecret,
+    remote_pub_key: x25519_dalek::PublicKey,
```

Comment:
nit: do you need to own these? ref would probably be enough.

Reply:
yep, ref is enough